woocommerce/woocommerce-ios
# REST API: Update `AlamofireNetwork` to support authentication with application passwords

Closes: #8389 


### Description

This PR updates the Networking layer to support authentication with application passwords:
- Created new request type `RESTRequest`. This requires a site URL to construct the endpoint URL for its path. It also accepts an optional fallback Jetpack request to be sent if application password is blocked for the given site.
- Created a helper class `RequestAuthenticator` to update requests with the appropriate Authorization header if possible. This class handles fetching application password for REST requests and wrapping other request types with WPCOM tokens.
- Updated `AlamofireNetwork` with a request authenticator to authenticate requests before sending them.
- ~Updated `DefaultStoresManager` and `AuthenticatedState` to send an updated site ID to `AlamofireNetwork` when the selected store changes - to update the application password use case accordingly.~

### Testing instructions

No API endpoint has been migrated to use REST API yet, but please run the app and do a smoke test to make sure that API requests still work as before.

### Screenshots

N/A

---
